crates/core/src/subscription/module_subscription_manager.rs

```diff
@@ -343,15 +552,122 @@ mod tests {
         let client = Arc::new(client(0));
 
         let mut subscriptions = SubscriptionManager::default();
-        subscriptions.add_subscription(client, [plan]);
+        subscriptions.set_legacy_subscription(client.clone(), [plan.clone()]);
 
         assert!(subscriptions.contains_query(&hash));
-        assert!(subscriptions.contains_subscription(&id, &hash));
+        assert!(subscriptions.contains_legacy_subscription(&id, &hash));
         assert!(subscriptions.query_reads_from_table(&hash, &table_id));
 
         Ok(())
     }
 
+    #[test]
+    fn test_subscribe_single_adds_table_mapping() -> ResultTest<()> {
+        let db = TestDB::durable()?;
+
+        let table_id = create_table(&db, "T")?;
+        let sql = "select * from T";
+        let plan = compile_plan(&db, sql)?;
+        let hash = plan.hash();
+
+        let client = Arc::new(client(0));
+
+        let query_id: ClientQueryId = QueryId::new(1);
+        let mut subscriptions = SubscriptionManager::default();
+        subscriptions.add_subscription(client.clone(), plan.clone(), query_id)?;
+        assert!(subscriptions.query_reads_from_table(&hash, &table_id));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_unsubscribe_from_the_only_subscription() -> ResultTest<()> {
```

Comment:
IIUC this will be a follow-up, as this PR contains only server-side handling. This test does not implement the client-side parts of the new API (see `todo`s in the `sdk` crate).

Reply:
I added a test with that case and a few others. I would also like to add tests that ensure we actually send updates, but I want to start writing tests covering the sdk side before I decide where to put those.